packages/focal/src/react/react.ts

```diff
@@ -15,111 +15,113 @@ export interface Subscription {
   unsubscribe(): void
 }
 
-export type Lifted<T> = {
-  [K in keyof T]: T[K] | Observable<T[K]>
-}
+export namespace LiftWrapper {
+  export type Lifted<T> = {
+    [K in keyof T]: T[K] | Observable<T[K]>
+  }
 
-export interface LiftWrapperProps<TProps> {
-  component: React.Component<TProps, any>
-    | React.StatelessComponent<TProps>
+  export interface Props<TProps> {
+    component: React.Component<TProps, any>
+    | React.FunctionComponent<TProps>
     | React.ComponentClass<TProps>
     | React.ComponentType
     | keyof React.ReactHTML
-  props: Lifted<TProps>
-}
+    props: Lifted<TProps>
+  }
 
-export interface LiftWrapperState {
-  renderCache?: React.DOMElement<any, any> | null
-  subscription?: Subscription | null
-}
+  export type RenderCache = React.DOMElement<any, any> | null
+  export type WrapperSubscription = Subscription | null
 
-/**
- * A wrapper component that allows to use observables for prop values of a
- * given component.
- */
-export class LiftWrapper<TProps>
-    extends React.Component<LiftWrapperProps<TProps>, LiftWrapperState> {
-  state = LiftWrapper._initState
+  export class Renderer<TProps> extends React.PureComponent<Props<TProps>> {
+    static displayName = 'LiftWrapper'
 
-  static _initState: LiftWrapperState = {
-    renderCache: null,
-    subscription: null
-  }
+    subscription: WrapperSubscription = null
+    renderCache: RenderCache = null
 
-  static _endState: LiftWrapperState = {
-    subscription: null
-  }
+    // variable to track sync emit from observable in Render{One, Many}.
+    // prevent waste render during initial render & re-subscribe
+    isSubscribed = false
 
-  render() {
-    return this.state.renderCache || null
-  }
+    // eslint-disable-next-line camelcase
+    UNSAFE_componentWillMount() {
+      this._subscribe(this.props)
```

Comment:
Is it possible to move `subscribe` into `constructor` to avoid dependency on legacy lifecycle?

In the article [Update on async rendering](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#fetching-external-data) React team provides two examples about pre-fetch and subscription.

> When supporting server rendering, it’s currently necessary to provide the data **synchronously** – componentWillMount was often used for this purpose but the **constructor** can be used as a replacement.

The [Adding event listener](https://reactjs.org/blog/2018/03/27/update-on-async-rendering.html#adding-event-listeners-or-subscriptions) suggest to use `componentDidMount` with initial value - it's not our case. But, [they ok with](url) using `componentDidUpdate` instead of `componentWillReceiveProps`

@KatSick what do you think?